src/System.Text.CaseFolding/System/Text/CaseFolding/SimpleCaseFolding.cs

```diff
@@ -0,0 +1,606 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Text.CaseFolding
+{
+    /// <summary>
+    /// </summary>
+    public static partial class SimpleCaseFolding
+
+    {
+        private static ref ushort s_MapLevel1 => ref MapLevel1[0];
+        private static ref char s_refMapData => ref MapData[0];
+        private static ref ushort s_refMapSurrogateLevel1 => ref MapSurrogateLevel1[0];
+        private static ref (char highSurrogate, char lowSurrogate) s_refMapSurrogateData => ref MapSurrogateData[0];
+
+        /// <summary>
+        /// </summary>
+        public static char SimpleCaseFold(char c)
```

Comment:
maybe adding [MethodImpl(MethodImplOptions.AggressiveInlining)] is worth here as you are calling this inside a loops later.

Reply:
Added.